include/geometry/LinesSet.h

```diff
@@ -10,6 +10,7 @@
 
 #include "geometry/OpenLinesSet.h"
 #include "geometry/Point2LL.h"
+#include "geometry/Polygon.h"
```

Comment:
I see no reason to include this header here (Polygon class is not even used in this header). Please include it only in the cpp files where it is actually required.